File: sdk/php/src/Resources/Ssl.php
<?php

namespace HiTechCloud\SDK\Resources;

class Ssl extends BaseResource
{
    /**
     * List certificates
     */
    public function list(array $params = []): array
    {
        return $this->http->get('/api/certificate', $params);
    }

    /**
     * Get certificate
     */
    public function get(int $certId): array
    {
        return $this->http->get('/api/certificate/' . $serviceId);
    }

    /**
     * Get certificate CRT
     */
    public function getCrt(int $certId): array
    {
        return $this->http->get('/api/certificate/' . $serviceId . '/crt');
    }

    /**
     * List SSL orders
     */
    public function listOrders(array $params = []): array
    {
        return $this->http->get('/api/certificate/order', $params);
    }

    /**
     * Create SSL order
     */
    public function createOrder(array $data): array
    {
        return $this->http->post('/api/certificate/order', $data);
    }

    /**
     * Get order software
     */
    public function getOrderSoftware(int $productId): array
    {
        return $this->http->get('/api/certificate/order/' . $productId . '/software');
    }

}

File: sdk/csharp/Resources/Ssl.cs
namespace HiTechCloud.SDK.Resources;

public class Ssl : BaseResource
{
    public Ssl(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> CertificatesAsync()
    {
        return await Http.GetAsync("/api/certificate");
    }

    public async Task<Dictionary<string, object?>> CertificateAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/certificate/{id}", id));
    }

    public async Task<Dictionary<string, object?>> CertificateCrtAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/certificate/{id}/crt", id));
    }

    public async Task<Dictionary<string, object?>> OrdersAsync()
    {
        return await Http.GetAsync("/api/certificate/order");
    }

    public async Task<Dictionary<string, object?>> CreateOrderAsync(Dictionary<string, object> data)
    {
        return await Http.PostAsync("/api/certificate/order", data);
    }

    public async Task<Dictionary<string, object?>> OrderSoftwareAsync(int productId)
    {
        return await Http.GetAsync(string.Format("/api/certificate/order/{productId}/software", productId));
    }
}
